test/unit/pdf_find_controller_spec.js

```diff
@@ -71,11 +71,23 @@ describe('pdf_find_controller', function() {
     pdfFindController = null;
   });
 
-  it('performs a basic search', function(done) {
-    pdfFindController.executeCommand('find', { query: 'Dynamic', });
+  function testSearch({ parameters, matchesPerPage, callback, }) {
```

Comment:
How about having this function return a `Promise` instead, and just remove the callback?

```javascript
function testSearch({ parameters, matchesPerPage, }) {
  return new Promise(function(resolve) {
    pdfFindController.executeCommand('find', parameters);

    // More code here...

    eventBus.on('updatefindmatchescount',
        function onUpdateFindMatchesCount(evt) {

      // Even more code here...

      resolve(); // Instead of callback();
  });
}
```

Then a callsite would just look like e.g.
```javascript
testSearch({
  parameters: {
    query: 'Dynamic',
    caseSensitive: false,
    entireWord: false,
    phraseSearch: true,
  },
  matchesPerPage: [11, 5, 0, 3, 0, 0, 0, 1, 1, 1, 0, 3, 4, 4],
}).then(done);
```

Reply:
Done.